nucleus/common/common-util/src/main/java/com/sun/enterprise/security/store/AsadminSecurityUtil.java

```diff
@@ -257,25 +257,29 @@ private void init(final char[] commandLineMasterPassword, final boolean isPrompt
             asadminTruststore = openTruststore(passwordToUse);
         } catch (IOException e) {
             logger.log(Level.WARNING, String.format("Error when reading truststore, exception:%s. Now recreating file", e));
-            recreateTrustStore(AsadminTruststore.getAsadminTruststore(), passwordToUse);
+            recreateDefaultTrustStore(passwordToUse);
         }
     }
 
     /**
-     * Method to recreate the truststore file
-     * @param fileTruststore File of the truststore
+     * Method to recreate the default truststore file from the .gfclient folder
+     *
      * @param passwordToUse password to use
      * @throws IOException
      */
-    private void recreateTrustStore(File fileTruststore, char[] passwordToUse) throws IOException {
-        try {
-            Files.deleteIfExists(fileTruststore.toPath());
-            asadminTruststore = openTruststore(passwordToUse);
-        } catch (IOException | KeyStoreException | NoSuchAlgorithmException | CertificateException e) {
-            logger.log(Level.WARNING,
-                    String.format("Error when processing truststore with path:%s and exception:%s",
-                            fileTruststore.toPath(), e));
-            throw new RuntimeException(e);
+    protected void recreateDefaultTrustStore(char[] passwordToUse) throws IOException {
+        if (System.getProperty(SystemPropertyConstants.CLIENT_TRUSTSTORE_PROPERTY) == null) {
```

Comment:
Shouldn't this check be before your log warning? The current flow is:

* Caught error trying to open store
* Log "Error when reading truststore, exception:%s. Now recreating file"
* Realise trust store isn't default and do nothing

Reply:
good catch, I already updated the condition to be before the whole recreate process